### PR TITLE
[BEAM-3545] Support Metrics in Go SDK CombineFns- Part 2

### PR DESCRIPTION
**Pull request by lostluck**
This PR cleans up a few things:
* transform.UniqueName appears to be -per bundle- in some instances, which defeats the point. Reverting until the runner I found that on changes behavior. Not caught earlier due to metrics being non-programmatically accessible at present.
* Fix a go lint on the "unset" constants I added. ID is prefered over Id.
* Actually add Combine PIDs to the list to pardos with which to export to the runner.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by lostluck** [comment]
R: @youngoli Serves me right for not doing additional testing on the PR, before review. It would have caught these issues sooner.

File: sdks/go/pkg/beam/core/runtime/exec/pardo.go [hunk 1]
```diff
@@ -79,6 +86,7 @@ func (n *ParDo) Up(ctx context.Context) error {
 	return nil
 }
 
+// StartBundle starts processing for this bundle and does before bundle processing operations.
```
Comment (youngoli) [inline sdks/go/pkg/beam/core/runtime/exec/pardo.go]: Sentence sounds awkward around "and does before bundle processing operations." Missing word?

Comment (youngoli) [reply sdks/go/pkg/beam/core/runtime/exec/pardo.go]: Nvm, just realized how this is meant to be read. Maybe hyphenate it so it says "pre-bundle-processing" to make it less ambiguous? It looks a bit funny like that, but it's preferable to being difficult to parse.

File: sdks/go/pkg/beam/core/runtime/exec/pardo.go [hunk 2]
```diff
@@ -148,6 +157,7 @@ func mustExplodeWindows(fn *funcx.Fn, elm *FullValue, usesSideInput bool) bool {
 	return explode || usesSideInput
 }
 
+// FinishBundle finishes processing for this bundle and does post bundle processing operations.
```
Comment (youngoli) [inline sdks/go/pkg/beam/core/runtime/exec/pardo.go]: Same as above comment. "post-bundle-processing"

File: sdks/go/pkg/beam/core/runtime/exec/pardo.go [hunk 3]
```diff
@@ -167,6 +177,7 @@ func (n *ParDo) FinishBundle(ctx context.Context) error {
 	return nil
 }
 
+// Down tearsdown DoFn resources.
```
Comment (youngoli) [inline sdks/go/pkg/beam/core/runtime/exec/pardo.go]: "tearsdown" should probably be two words.

File: sdks/go/pkg/beam/core/runtime/exec/pardo.go (hunk 1 above)
Comment (lostluck) [reply sdks/go/pkg/beam/core/runtime/exec/pardo.go]: Cleaned it up. Arguably a bundle hasn't "started" until after this method returns.

File: sdks/go/pkg/beam/core/runtime/exec/pardo.go (hunk 3 above)
Comment (lostluck) [reply sdks/go/pkg/beam/core/runtime/exec/pardo.go]: Done.

File: sdks/go/pkg/beam/core/runtime/exec/pardo.go (hunk 2 above)
Comment (lostluck) [reply sdks/go/pkg/beam/core/runtime/exec/pardo.go]: Went without the bundle-processing hyphen, since post-bundle is the adverb for the verb processing.

**Comment by lostluck** [comment]
Thanks for the review!

R: @aaltay Please review and merge.

File: sdks/go/pkg/beam/core/runtime/exec/pardo.go (hunk 2 above)
Comment (youngoli) [reply sdks/go/pkg/beam/core/runtime/exec/pardo.go]: Oh yeah, that looks way better than my suggestion :+1:

**Pull request merged**